core/src/main/java/io/methvin/watcher/hashing/Hash.java

```diff
@@ -0,0 +1,9 @@
+package io.methvin.watcher.hashing;
+
+public interface Hash {
```

Comment:
A couple thoughts:
 1. From `DirectoryWatcher`s perspective, all it needs is `equals`, so we don't actually _need_ any other methods in the interface—though we may want them for other reasons.
 2. Do you think getting the hash as a byte array is actually useful to the end user? What if the user wants to create a custom `FileHasher` with an alternative implementation of `Hash` that's based on a `long` or a `String` or some other type?
 3. How would you imagine `asString` differs from `toString`? I could see `toString` encoding other information about the specific hash information, but `asString` could also do the same.